internal/lang/funcs/filesystem_test.go

```diff
@@ -217,6 +217,63 @@ func TestTemplateFile(t *testing.T) {
 	}
 }
 
+func TestTemplateString(t *testing.T) {
+	tests := []struct {
+		Data  cty.Value
+		Vars  cty.Value
+		Want  cty.Value
+		Err   string
+	}{
+		{
+			cty.StringVal("Hello, Jodie!"),
+			cty.EmptyObjectVal,
+			cty.StringVal("Hello, Jodie!"),
+			``,
+		},
+		{
+			cty.StringVal("Hello, ${name}!"),
+			cty.MapVal(map[string]cty.Value{
+				"name": cty.StringVal("Jodie"),
+			}),
+			cty.StringVal("Hello, Jodie!"),
+			``,
+		},
+		// Add more test cases for templatestring here
```

Comment:
Can you add the missing tests please, this seems to only be the happy path.

Reply:
commit [4fb1950](https://github.com/opentofu/opentofu/pull/1223/commits/4fb1950bb377f66c61af3af40afb85f8d11d343a) resolves this